### ISSUE TITLE
In `read_csv`, using `infer_schema_length <= skip_rows_after_header` causes all `dtypes` to be `string`

### ISSUE THREAD

**Issue by niksubramanian** [pola-rs/polars#7698]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

In polars `read_csv()` and `scan_csv()` functions, using`infer_schema_length <= skip_rows_after_header` causes all `dtypes` to be `string`, rather than the correct dtype.

PS: thank you for polars - it really is a great library.

### Reproducible example

```python
import polars as pl


# create a dataframe with an `int` column and a `float` column.
pl.DataFrame(
    {
        "a": [1, 2, 3, 4, 5],
        "b": [0.1, 0.2, 0.3, 0.4, 0.5],
    }
).write_csv("test.csv")

# ---------
# buggy behavior
# ---------
df = pl.read_csv(
    "test.csv",
    infer_schema_length=2,
    skip_rows_after_header=2,
)

# The dataframe sets all columns' dtypes to `str`, rather than the correct types.
print(df)
>>> ┌─────┬─────┐
>>> │ a   ┆ b   │
>>> │ --- ┆ --- │
>>> │ str ┆ str │
>>> ╞═════╪═════╡
>>> │ 3   ┆ 0.3 │
>>> │ 4   ┆ 0.4 │
>>> │ 5   ┆ 0.5 │
>>> └─────┴─────┘

# ---------
# correct behavior
# ---------
df = pl.read_csv(
    "test.csv",
    infer_schema_length=2,
    skip_rows_after_header=1,
)

# The dataframe sets all columns' dtypes to `str`, rather than the correct types.
print(df)
>>> ┌─────┬─────┐
>>> │ a   ┆ b   │
>>> │ --- ┆ --- │
>>> │ i64 ┆ f64 │
>>> ╞═════╪═════╡
>>> │ 2   ┆ 0.2 │
>>> │ 3   ┆ 0.3 │
>>> │ 4   ┆ 0.4 │
>>> │ 5   ┆ 0.5 │
>>> └─────┴─────┘
```

### Expected behavior

infer_schema_length should use the dtype of the rows after the skipped rows to infer the schema.

### Installed versions

<details>

```
---Version info---
Polars: 0.16.14
Index type: UInt32
Platform: Linux-6.2.7-arch1-1-x86_64-with-glibc2.37
Python: 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:35:26) [GCC 10.4.0]
---Optional dependencies---
numpy: 1.23.3
pandas: 1.4.4
pyarrow: 9.0.0
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: 3.5.3
xlsx2csv: <not installed>
xlsxwriter: <not installed>
None
```

</details>

**Comment by ghuls** [comment]
Got it working in my patched local version:

```python
In [8]: df = pl.read_csv(
   ...:     "test.csv",
   ...:     infer_schema_length=2,
   ...:     skip_rows_after_header=2,
   ...: )

In [9]: df
Out[9]: 
shape: (3, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ i64 ┆ f64 │
╞═════╪═════╡
│ 3   ┆ 0.3 │
│ 4   ┆ 0.4 │
│ 5   ┆ 0.5 │
└─────┴─────┘

In [10]: df = pl.read_csv(
    ...:     "test.csv",
    ...:     infer_schema_length=2,
    ...:     skip_rows_after_header=3,
    ...: )

In [11]: df
Out[11]: 
shape: (2, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ i64 ┆ f64 │
╞═════╪═════╡
│ 4   ┆ 0.4 │
│ 5   ┆ 0.5 │
└─────┴─────┘
```

**Comment by ghuls** [comment]
Closed by #7701
